tests/test_dataset.py

```diff
@@ -0,0 +1,296 @@
+        dirname, "../testoutput/tiff_slice_dataset_out_of_bounds/"
+    )
+    mag_filename = path.join(new_dataset_path, "color/1")
+
+    delete_dir(new_dataset_path)
+
+    copytree(old_dataset_path, new_dataset_path)
+
+    slice = TiffSlice(mag_filename, size=(100, 100, 100))
+
+    with slice.open():
+        try:
+            slice.write(
+                np.zeros((200, 200, 5), dtype=np.uint8)
+            )  # this is bigger than the bounding_box
+            raise AssertionError(
+                "The test 'test_slice_tiff_write_out_of_bounds' did not throw an exception even though it should"
+            )
+        except Exception:
+            pass
+
+
+def test_tiff_write_out_of_bounds():
+    dirname = path.dirname(__file__)
+    old_dataset_path = path.join(dirname, "../testdata/simple_tiff_dataset/")
+    new_dataset_path = path.join(
+        dirname, "../testoutput/simple_tiff_dataset_out_of_bounds/"
+    )
+
+    delete_dir(new_dataset_path)
+
+    copytree(old_dataset_path, new_dataset_path)
+
+    ds = TiffDataset.open(new_dataset_path)
+    mag_dataset = ds.get_layer("color").get_mag("1")
+
+    assert ds.properties.data_layers["color"].get_bounding_box_size() == (265, 265, 10)
+    mag_dataset.write(
+        np.zeros((300, 300, 15), dtype=np.uint8)
+    )  # this is bigger than the bounding_box
+    assert ds.properties.data_layers["color"].get_bounding_box_size() == (300, 300, 15)
+
+
+def test_wk_write_out_of_bounds():
+    dirname = path.dirname(__file__)
+    old_dataset_path = path.join(dirname, "../testdata/simple_wk_dataset/")
+    new_dataset_path = path.join(
+        dirname, "../testoutput/simple_wk_dataset_out_of_bounds/"
+    )
+
+    delete_dir(new_dataset_path)
+
+    copytree(old_dataset_path, new_dataset_path)
+
+    ds = WKDataset.open(new_dataset_path)
+    mag_dataset = ds.get_layer("color").get_mag("1")
+
+    assert ds.properties.data_layers["color"].get_bounding_box_size() == (
+        1024,
+        1024,
+        1024,
+    )
+    mag_dataset.write(
+        np.zeros((3, 1, 1, 2048), dtype=np.uint8)
+    )  # this is bigger than the bounding_box
+    assert ds.properties.data_layers["color"].get_bounding_box_size() == (
+        1024,
+        1024,
+        2048,
+    )
+
+
+def test_tiff_write_multi_channel_uint8():
+    dirname = path.dirname(__file__)
+    dataset_path = path.join(dirname, "../testoutput/tiff_multichannel/")
+
+    delete_dir(dataset_path)
+
+    ds_tiff = TiffDataset.create(dataset_path, [1])
+    ds_tiff.add_layer("color", "color", num_channels=3)
+    ds_tiff.get_layer("color").add_mag("1")
+
+    # 10 images (z-layers), each 250x250, dtype=np.uint8
+    data = np.zeros((3, 250, 250, 10), dtype=np.uint8)
+    for h in range(10):
+        for i in range(250):
+            for j in range(250):
+                data[0, i, j, h] = i
+                data[1, i, j, h] = j
+                data[2, i, j, h] = 100
+
+    ds_tiff.get_layer("color").get_mag("1").write(data)
```

Comment:
There is no assertion in this test. Maybe just read out again the same bounding box and assert that the output is equal to `data`?